### DISCUSSION TITLE
Can I run Cypress tests on Iphone and Android using different browsers?

### DISCUSSION THREAD

**Discussion by punisher49** [cypress-io/cypress#15729] [Questions and Help] [answered]
Can anybody let me know if it is actually possible to run Cypress tests on real phone or on Android Studio emulator and Xcode simulator?

**Comment by jennifer-shehane** [comment, marked as answer]
This is not currently possible to do in Cypress. You can test web application at different viewport sizes, but Cypress must be run in a desktop environment currently.

**Reply by brobee** [reply]
Hi Jennifer! I set my user-agent in my cypress.json and no other modification during runtime. However I call the cy.visit twice, first one is for authentication/login in before and the second one is the actual test. The baseURL is the same, but the user agent is set for the first cy.visit only. Second one is reverted back to my os userAgent. Is there any solution for this? Thanks!
